ansible/roles/passwords/defaults/main.yml

```diff
@@ -1,16 +1,14 @@
 ---
 
-openhpc_passwords:
-  secrets_openhpc_grafana_admin_password: "{{ secrets_openhpc_grafana_admin_password | default(lookup('password', '/dev/null')) }}"
-  secrets_openhpc_elasticsearch_admin_password: "{{ secrets_openhpc_elasticsearch_admin_password | default(lookup('password', '/dev/null')) }}"
-  secrets_openhpc_elasticsearch_kibana_password: "{{ secrets_openhpc_elasticsearch_kibana_password | default(lookup('password', '/dev/null')) }}"
-  secrets_openhpc_mysql_root_password: "{{ secrets_openhpc_mysql_root_password | default(lookup('password', '/dev/null')) }}"
-  secrets_openhpc_mysql_slurm_password: "{{ secrets_openhpc_mysql_slurm_password | default(lookup('password', '/dev/null')) }}"
-  # TODO: Might be cleaner to write this to a file, but needs modification of the role
-  secrets_openhpc_mungekey: "{{ secrets_openhpc_mungekey | default(secrets_openhpc_mungekey_default) }}"
+slurm_appliance_secrets:
+  vault_grafana_admin_password: "{{ secrets_openhpc_grafana_admin_password | default(vault_grafana_admin_password | default(lookup('password', '/dev/null'))) }}"
```

Comment:
Just wondering if we need to set boolean=true[1] in second call to the default filter as the value coming out of the first will be the empty string i.e default(value, true). Although if you've tested this and it works ignore this comment.

[1] https://jinja.palletsprojects.com/en/3.0.x/templates/#jinja-filters.default

Reply:
Interesting. It does work as-is.